specs/lib/moveTechniqueRenderStatesSpec.js

```diff
@@ -0,0 +1,219 @@
+'use strict';
+var Cesium = require('cesium');
+var moveTechniqueRenderStates = require('../../lib/moveTechniqueRenderStates');
+
+var WebGLConstants = Cesium.WebGLConstants;
+
+var gltf = {
+    programs: {
+        program_0: {
+            attributes: [
+                'a_normal',
+                'a_position',
+                'a_texcoord0'
+            ],
+            fragmentShader: 'BoxTextured0FS',
+            vertexShader: 'BoxTextured0VS'
+        }
+    },
+    shaders: {
+        BoxTextured0FS: {
+            type: WebGLConstants.FRAGMENT_SHADER,
+            uri: 'BoxTextured0FS.glsl'
+        },
+        BoxTextured0VS: {
+            type: WebGLConstants.VERTEX_SHADER,
+            uri: 'BoxTextured0VS.glsl'
+        }
+    },
+    techniques: {
+        technique0: {
+            states: {
+                enable: [
+                    WebGLConstants.DEPTH_TEST,
+                    WebGLConstants.BLEND,
+                    WebGLConstants.CULL_FACE
+                ]
+            }
+        }
+    },
+    materials: [
+        {
+            technique: 'technique0'
+        }
+    ]
+};
+
+describe('updateMaterialProperties', function() {
+    it('sets material.doubleSided property if CULL_FACE is not enabled', function () {
+        var gltfWithUpdatedMaterials = moveTechniqueRenderStates(gltf);
+        var material = gltfWithUpdatedMaterials.materials[0];
+        expect(material.doubleSided).toBeUndefined();
+
+        var gltfDoubleSided = JSON.parse(JSON.stringify(gltf));
```

Comment:
Maybe use Cesium's `clone` function instead. `clone(gltf, true)`.

Reply:
I was having issues with the depth of the cloning when it came to arrays (even when using the deep clone argument). This was much more reliable.

Reply:
Is that a Cesium bug? I don't think that should be happening...

Reply:
The documentation isn't partially clear on what the behavior should be: https://cesiumjs.org/Cesium/Build/Documentation/clone.html

Reply:
Alright, if this way works consistently then let's go with it.